Build with VS 2017 only doesn't work

I followed the instructions mentioned here: https://github.com/dotnet/coreclr/blob/master/Documentation/building/windows-instructions.md for Visual Studio 2017.

I don't have VS 2015.

When calling build from Developer Command Prompt  for VS 2017 I receive this error:

```
Running: C:\Users\myuser\Source\coreclr\Tools\dotnetcli\dotnet.exe C:\Users\myuser\Source\coreclr\Tools\run.exe C:\Users\myuser\Source\coreclr\config.json build -Project=C:\Users\myuser\Source\coreclr\build.proj -generateHeaderWindows -NativeVersionHeaderFile="C:\Users\myuser\Source\coreclr\bin\obj\_version.h" -BuildOS=Windows_NT -BuildType=Debug -BuildArch=x64
Running: C:\Users\myuser\Source\coreclr\Tools\msbuild.cmd /nologo /verbosity:minimal /clp:Summary /maxcpucount /nodeReuse:false  /l:BinClashLogger,Tools/net46/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log  C:\Users\myuser\Source\coreclr\build.proj /p:__BuildType=Debug /p:__BuildArch=x64 /p:__BuildOS=Windows_NT  /t:GenerateVersionHeader /p:GenerateVersionHeader=true /p:NativeVersionHeaderFile=C:\Users\myuser\Source\coreclr\bin\obj\_version.h
MSBUILD : error MSB1021: Cannot create an instance of the logger. Could not load file or assembly 'Tools\/net46\/Microsoft.DotNet.Build.Tasks.dll' or one of its dependencies. The given assembly name or codebase was invalid. (Exception from HRESULT: 0x80131047)
Switch: BinClashLogger,Tools/net46/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log
Command execution failed with exit code 1.
BUILD: Restoring the OptimizationData Package
Tools are already initialized.
Running: C:\Users\myuser\Source\coreclr\Tools\dotnetcli\dotnet.exe C:\Users\myuser\Source\coreclr\Tools\run.exe C:\Users\myuser\Source\coreclr\config.json sync -optdata
Running: C:\Users\myuser\Source\coreclr\Tools\msbuild.cmd /nologo /verbosity:minimal /clp:Summary /maxcpucount /nodeReuse:false  /flp:v=detailed;LogFile=sync.log /t:RestoreOptData /p:RestoreDuringBuild=true  ./build.proj
MSBUILD : error MSB1009: Project file does not exist.
Switch: ./build.proj
Command execution failed with exit code 1.
BUILD: Error: Failed to restore the optimization data package.
```

I'm having this issue also. From VS2017 developer command prompt:

```
C:\Users\jmusser\Source\Repos\coreclr>build
Starting Build at 21:36:58.97
BUILD: Commencing CoreCLR Repo build
BUILD: Checking prerequisites
BUILD: Using environment: "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\Tools\\VsDevCmd.bat"
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.0.26403.7
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
Installing dotnet cli...
Restoring BuildTools version 1.0.27-prerelease-01413-01...
Initializing BuildTools ...
Updating CLI NuGet Frameworks map...
Done initializing tools.
Running: C:\Users\jmusser\Source\Repos\coreclr\Tools\dotnetcli\dotnet.exe C:\Users\jmusser\Source\Repos\coreclr\Tools\run.exe C:\Users\jmusser\Source\Repos\coreclr\config.json build -Project=C:\Users\jmusser\Source\Repos\coreclr\build.proj -generateHeaderWindows -NativeVersionHeaderFile="C:\Users\jmusser\Source\Repos\coreclr\bin\obj\_version.h" -BuildOS=Windows_NT -BuildType=Debug -BuildArch=x64
Running: C:\Users\jmusser\Source\Repos\coreclr\Tools\msbuild.cmd /nologo /verbosity:minimal /clp:Summary /maxcpucount /nodeReuse:false  /l:BinClashLogger,Tools/net46/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log  C:\Users\jmusser\Source\Repos\coreclr\build.proj /p:__BuildType=Debug /p:__BuildArch=x64 /p:__BuildOS=Windows_NT  /t:GenerateVersionHeader /p:GenerateVersionHeader=true /p:NativeVersionHeaderFile=C:\Users\jmusser\Source\Repos\coreclr\bin\obj\_version.h
MSBUILD : error MSB1021: Cannot create an instance of the logger. Could not load file or assembly 'Tools\/net46\/Microsoft.DotNet.Build.Tasks.dll' or one of its dependencies. The given assembly name or codebase was invalid. (Exception from HRESULT: 0x80131047)
Switch: BinClashLogger,Tools/net46/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log
Command execution failed with exit code 1.
```

https://github.com/dotnet/coreclr/issues/10181 looks related.

Here is the log from the brand new Windows install:

```
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.0.26403.7
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************

C:\Users\felip\Source>cd C:\Users\felip\Source\coreclr

C:\Users\felip\Source\coreclr>build skiptests
Starting Build at 23:24:09.54
BUILD: Commencing CoreCLR Repo build
BUILD: Checking prerequisites
BUILD: Using environment: "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\Tools\\VsDevCmd.bat"
**********************************************************************
** Visual Studio 2017 Developer Command Prompt v15.0.26403.7
** Copyright (c) 2017 Microsoft Corporation
**********************************************************************
Installing dotnet cli...
Restoring BuildTools version 1.0.27-prerelease-01413-01...
Initializing BuildTools ...
Updating CLI NuGet Frameworks map...
Done initializing tools.
Running: C:\Users\felip\Source\coreclr\Tools\dotnetcli\dotnet.exe C:\Users\felip\Source\coreclr\Tools\run.exe C:\Users\felip\Source\coreclr\config.json build -Project=C:\Users\felip\Source\coreclr\build.proj -generateHeaderWindows -NativeVersionHeaderFile="C:\Users\felip\Source\coreclr\bin\obj\_version.h" -BuildOS=Windows_NT -BuildType=Debug -BuildArch=x64
Running: C:\Users\felip\Source\coreclr\Tools\msbuild.cmd /nologo /verbosity:minimal /clp:Summary /maxcpucount /nodeReuse:false  /l:BinClashLogger,Tools/net46/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log  C:\Users\felip\Source\coreclr\build.proj /p:__BuildType=Debug /p:__BuildArch=x64 /p:__BuildOS=Windows_NT  /t:GenerateVersionHeader /p:GenerateVersionHeader=true /p:NativeVersionHeaderFile=C:\Users\felip\Source\coreclr\bin\obj\_version.h
MSBUILD : error MSB1021: Cannot create an instance of the logger. Could not load file or assembly 'Tools\/net46\/Microsoft.DotNet.Build.Tasks.dll' or one of its dependencies. The given assembly name or codebase was invalid. (Exception from HRESULT: 0x80131047)
Switch: BinClashLogger,Tools/net46/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log
Command execution failed with exit code 1.
BUILD: Restoring the OptimizationData Package
Tools are already initialized.
Running: C:\Users\felip\Source\coreclr\Tools\dotnetcli\dotnet.exe C:\Users\felip\Source\coreclr\Tools\run.exe C:\Users\felip\Source\coreclr\config.json sync -optdata
Running: C:\Users\felip\Source\coreclr\Tools\msbuild.cmd /nologo /verbosity:minimal /clp:Summary /maxcpucount /nodeReuse:false  /flp:v=detailed;LogFile=sync.log /t:RestoreOptData /p:RestoreDuringBuild=true  ./build.proj
MSBUILD : error MSB1009: Project file does not exist.
Switch: ./build.proj
Command execution failed with exit code 1.
BUILD: Error: Failed to restore the optimization data package.
```

@tannergooding Running the failing command gives the original error. Running the command you corrected gives:

```
MSBUILD : error MSB1021: Cannot create an instance of the logger. The given assembly name or codebase was invalid. (Exception from HRESULT: 0x80131047)
Switch: BinClashLogger,Tools\net46\Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log
```

Guess I'll post a diagnostic log next.

--- a/run.cmd
+++ b/run.cmd
@@ -53,9 +53,15 @@ xcopy %~dp0Tools-Override\* %~dp0Tools /y >nul
 
 set _toolRuntime=%~dp0Tools
 set _dotnet=%_toolRuntime%\dotnetcli\dotnet.exe
+set _json=%~dp0config.json
 
+:: run.exe depends on running in the root directory, notably because the config.json specifies
+:: a relative path to the binclash logger
+
+pushd %~dp0
 echo Running: %_dotnet% %_toolRuntime%\run.exe %~dp0config.json %*
-call %_dotnet% %_toolRuntime%\run.exe %~dp0config.json %*
+call %_dotnet% %_toolRuntime%\run.exe "%_json%" %*
+popd
 if NOT [%ERRORLEVEL%]==[0] (
   exit /b 1
 )

